core-bundle/src/Image/PictureFactory.php

```diff
@@ -148,7 +148,8 @@ public function create($path, $size = null, ResizeOptions $options = null): Pict
      *
      * @param int|array|null $size
      *
-     * @return array<(PictureConfiguration|array<string, string>|ResizeOptions|null)>
+     * @return array<(PictureConfiguration|array<string, string>|ResizeOptions)>
+     * @psalm-return array{0:PictureConfiguration, 1:array<string, string>, 2:ResizeOptions}
```

Comment:
I am not yet sure how to handle this case. In PHPUnit, if there is a return type, the `@return` is omitted entirely.

```php
    /**
     * Creates a picture configuration.
     *
     * @param int|array|null $size
     *
     * @psalm-return array{0:PictureConfiguration, 1:array<string, string>, 2:ResizeOptions}
     */
```

If we decide to use both, do we group all `@psalm-` tags and `@phpstan-` tags or do we add the `@*-param` tags to the `@param` tags and the `@*-return` tags to the `@return` tags and so on?

```php
    /**
     * Creates a picture configuration.
     *
     * @param int|array|null $size
     * @phpstan-param …
     * @psalm-param …
     *
     * @return …
     * @phpstan-return …
     * @psalm-return array{0:PictureConfiguration, 1:array<string, string>, 2:ResizeOptions}
     */
```

Or rather:

```php
    /**
     * Creates a picture configuration.
     *
     * @param int|array|null $size
     *
     * @phpstan-param …
     * @phpstan-return …
     *
     * @psalm-param …
     * @psalm-return array{0:PictureConfiguration, 1:array<string, string>, 2:ResizeOptions}
     *
     * @return …
     */
```

Reply:
Your first grouping example imo looks more semantical to me.

I think we can at least omit either one of the `phpstan` or `psalm` prefixed tags as both tools read both. Imo we should keep the `@return` tags if they provide additional type information for the IDEs.

Reply:
I like the second one better (all `@phpstan-` and all `@psalm-` tags in separate groups). I agree that we might not need both, though.

> Imo we should keep the @return tags if they provide additional type information for the IDEs.

And is this the case here?

Reply:
Right. In this special case the IDE already knows the types because each of the destructured variable's types in `[$config, $attributes, $options] = $this->createConfig($size)` can already be inferred independent of the function.